.github/workflows/go.yaml

```diff
@@ -8,29 +8,43 @@ on:
   workflow_dispatch: # Enables the web UI manual trigger
   # Note: ^^ will fail the "tag-release" step if the git tag already exists
 
+concurrency:
+  group: ci-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   test-go:
-    name: Test, Vet, Vulncheck Go code
-    runs-on: ubuntu-22.04
+    name: Test Go
+    runs-on: ubuntu-latest
     timeout-minutes: 15
+
     strategy:
+      fail-fast: false
+      # This is a special matrix that tests against the listed version AND the
+      # Go version from go.mod. See usage in the actions/setup-go step.
       matrix:
-        go-version: ["1.22", "1.23", "1.24"]
-    env:
-      CGO_ENABLED: 0
+        include:
+          - go-version: "stable"
+          - go-version-file: "go.mod"
+
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4.2.2
         with:
           submodules: recursive
 
-      - name: Set up Go
+      - name: Set up Go Version
         uses: actions/setup-go@v5.5.0
         with:
           go-version: ${{ matrix.go-version }}
-          # check-latest: true # not needed when using 'stable'
+          go-version-file: ${{ matrix.go-version-file }}
```

Comment:
I'm a bit confused how this works?

We tell `setup-go` to use stable and then straight after we tell it to use go.mod

Won't one override the other?

**EDIT:** Looks like that is what happens https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

> If both the go-version and the go-version-file inputs are provided then the go-version input is used.

Reply:
Yes, this is intentional that it overrides only in some of the matrix cases.